doc/contributing/submitting-changes.chapter.md

```diff
@@ -214,47 +214,87 @@ The last checkbox is fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blo
 - Hydra builds for master and staging should not be used as testing platform, it’s a build farm for changes that have been already tested.
 - When changing the bootloader installation process, extra care must be taken. Grub installations cannot be rolled back, hence changes may break people’s installations forever. For any non-trivial change to the bootloader please file a PR asking for review, especially from \@edolstra.
 
+### Branches
+
+The nixpkgs repo has three major branches: `master`, `staging`, and
+`staging-next`.
+
+The most important distinction between them is that `staging`
+(colored red in the diagram below) can receive commits which cause
+a mass-rebuild (for example, anything that changes the `drvPath` of
+`stdenv`).  The other two branches `staging-next` and `master`
+(colored green in the diagram below) can *not* receive commits which
```

Comment:
Colors are transient and not even visible to everyone, depending on output medium or anatomy. Also another place for double book keeping to keep the diagram and the text in sync.

We should clarify what a mass rebuild is without going into implementation details that aren't relevant here. Tried to incorporate that into the suggestion.

```suggestion
can receive commits that cause
a mass-rebuild, that is, changes to data a large amount of derivations depend on and therefore require those to be rebuilt.
The other two branches, `staging-next` and `master`, must *not* receive commits which
```

Reply:
Yes, I'm aware that colorblind people exist, which is one of the reasons why this PR switches from *pastels* to primary colors.

Mentioning the fact that certain parts of the diagram are certain colors does not hurt, even if a few people cannot see those colors.